vtex/faststore
# Feat: Image using next/future/image + new Thumbor URL

## What's the purpose of this pull request?

- [x] uses new thumbor domain (vtexassets) - https://www.faststore.dev/releases/2023/03/10/faststore
- [x] adds sharp lib (production webp optimization - https://nextjs.org/docs/messages/sharp-missing-in-production) 
- [x] adds new thumbor URL as images domains https://nextjs.org/docs/basic-features/image-optimization#domains
- [x] uses next/future/image as default FS image component.
- [x] removes unnecessary styles.
- [x] tweaks Image Gallery Selectors ref and styles.


## How to test it?

- you can run `yarn build && yarn serve` locally and test the images from all the pages. Check Devtools tab, filter img and double-check the domain, type, size, and time for each image, along with the cache from the response headers.
(expected webp format and HIT the Cloudfront cache)
- you can use the starter PR and check if the images are working as expected webp format and HIT the Cloudfront cache in Devtols response Headers.

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/18

## References

https://nextjs.org/docs/messages/sharp-missing-in-production
https://nextjs.org/docs/api-reference/next/image#loader
https://nextjs.org/docs/basic-features/image-optimization#loaders
https://nextjs.org/docs/api-reference/next/image#loader-configuration
https://nextjs.org/docs/basic-features/image-optimization#domains
https://nextjs.org/docs/api-reference/next/image#caching-behavior
https://help.vtex.com/en/announcements/na-vtex-conversao-de-png-para-webp-agora-e-lossless--4J4ZtrCZIksY8qikWgeomY
